app/src/main/java/app/grapheneos/camera/util/CameraControl.kt

```diff
@@ -0,0 +1,28 @@
+package app.grapheneos.camera.util
+
+import androidx.camera.core.ZoomState
+import app.grapheneos.camera.CamConfig
+
+class CameraControl(private val camConfig: CamConfig) {
+
+    private fun zoomState(): ZoomState? = camConfig.camera?.cameraInfo?.zoomState?.value
+
+    fun zoomIn() = zoomByRatio(1f)
+
+    fun zoomOut() = zoomByRatio(-1f)
+
+    private fun zoomByRatio(zoomValue: Float) {
+        val zoomState = zoomState() ?: return
+        val currentZoomRatio = zoomState.zoomRatio
+        val newZoomRatio = currentZoomRatio + zoomValue
+
+        val zoomTo =
+            if (newZoomRatio > zoomState.maxZoomRatio) zoomState.maxZoomRatio //don't go above max zoom ratio
```

Comment:
These 2 comments about max zoom ratio and min zoom ratio really aren't needed since the code is very clear. Also, the one below about min zoom ratio still says max.